pkg/util/pki/csr.go

```diff
@@ -250,7 +264,7 @@ func GenerateCSR(crt *v1.Certificate, optFuncs ...GenerateCSROption) (*x509.Cert
 		ExtraExtensions:    extraExtensions,
 	}
 
-	if isLiteralCertificateSubjectEnabled() && len(crt.Spec.LiteralSubject) > 0 {
+	if opts.UseLiteralSubject && len(crt.Spec.LiteralSubject) > 0 {
```

Comment:
Do we actually have to gate the use of this `LiteralSubject` field? I think this could be a validation webhook-only gate right?